identity-service/src/routes/userBankTransactionMetadata.js

```diff
@@ -0,0 +1,94 @@
+const models = require('../models')
+const {
+  handleResponse,
+  successResponse,
+  errorResponseBadRequest,
+  errorResponseServerError
+} = require('../apiHelpers')
+const { logger } = require('../logging')
+const { Op } = require('sequelize')
+const { Validator } = require('jsonschema')
+const authMiddleware = require('../authMiddleware')
+
+const requestSchema = {
+  id: '/transaction_metadata_post',
+  type: 'object',
+  properties: {
+    transactionSignature: { type: 'string' },
+    metadata: { $ref: '/PurchaseMetadata' }
+  },
+  additionalProperties: false,
+  required: ['transactionSignature', 'metadata']
+}
+
+const purchaseMetadataSchema = {
+  id: '/PurchaseMetadata',
+  type: 'object',
+  properties: {
+    discriminator: { type: 'string' },
+    usd: { type: 'string' },
+    sol: { type: 'string' },
+    audio: { type: 'string' },
+    swapTransaction: { type: 'string' },
+    buyTransaction: { type: 'string' }
+  },
+  additionalProperties: false,
+  required: ['discriminator', 'usd', 'sol', 'audio', 'swapTransaction', 'buyTransaction']
+}
+
+const validator = new Validator()
+validator.addSchema(purchaseMetadataSchema)
+
+module.exports = function (app) {
+  app.post(
+    '/transaction_metadata',
+    authMiddleware,
+    handleResponse(async (req, res, next) => {
+      const userId = req.user.blockchainUserId
```

Comment:
I think we need to confirm that the auth'd userId here matches the userId in the request body

Reply:
we don't have a userId in the request body?